pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/when/impl/ChangeSetConditional.java

```diff
@@ -25,48 +25,96 @@
 
 package org.jenkinsci.plugins.pipeline.modeldefinition.when.impl;
 
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
 import hudson.Extension;
+import hudson.scm.ChangeLogSet;
+import hudson.util.ListBoxModel;
 import org.codehaus.groovy.ast.expr.Expression;
 import org.jenkinsci.Symbol;
 import org.jenkinsci.plugins.pipeline.modeldefinition.ast.ModelASTWhenContent;
 import org.jenkinsci.plugins.pipeline.modeldefinition.parser.ASTParserUtils;
 import org.jenkinsci.plugins.pipeline.modeldefinition.when.DeclarativeStageConditional;
 import org.jenkinsci.plugins.pipeline.modeldefinition.when.DeclarativeStageConditionalDescriptor;
+import org.jenkinsci.plugins.pipeline.modeldefinition.when.utils.Comparator;
 import org.kohsuke.stapler.DataBoundConstructor;
 import org.kohsuke.stapler.DataBoundSetter;
 
 import javax.annotation.CheckForNull;
 import javax.annotation.Nonnull;
+import java.io.IOException;
 
 /**
  * Conditional that checks the affected file paths recorded in the changelog.
  *
  * The build must first have collected the changelog via for example <code>checkout scm</code>.
  */
+@SuppressFBWarnings(value = "SE_NO_SERIALVERSIONID")
 public class ChangeSetConditional extends DeclarativeStageConditional<ChangeSetConditional> {
 
-    private String glob;
+    @Deprecated
+    private transient String glob;
+    private String pattern;
     private boolean caseSensitive;
+    private String comparator;
 
     @DataBoundConstructor
-    public ChangeSetConditional(String glob) {
-        this.glob = glob;
+    public ChangeSetConditional(String pattern) {
+        this.pattern = pattern;
         this.caseSensitive = false;
     }
 
+    @Deprecated
     public String getGlob() {
         return glob;
     }
 
+    @Deprecated
```

Comment:
The problem here is that we still have `caseSensitive` in the UI and there may be code out there using it as well.  
Can you leave it in?  people probably still want the option for case insensitive globs.

Reply:
Sure thing. I thought it might be easier to deprecate it and encourage people to use regexp.

Reply:
See https://github.com/jenkinsci/pipeline-model-definition-plugin/pull/366/commits/d138068ca09c45b8e235350a039bf2c913cb3b49 in conjunction with https://github.com/jenkinsci/pipeline-model-definition-plugin/pull/366/commits/2e17299977deb9b102caa3983eec019196ed0e89 where I added the support to the case sensitive for the GLOB comparator.